service.php

```diff
@@ -250,13 +253,18 @@ function ( $match ) use ( $dirpath ) {
 		}
 	}
 
-	header( 'Last-Modified: ' . gmdate( 'D, d M Y H:i:s', $last_modified ) . ' GMT' );
-	header( 'Content-Length: ' . ( strlen( $pre_output ) + strlen( $output ) ) );
-	header( "Content-Type: $mime_type" );
+	$headers = array(
+		'Last-Modified: ' . gmdate( 'D, d M Y H:i:s', $last_modified ) . ' GMT',
+		'Content-Length: ' . ( strlen( $pre_output ) + strlen( $output ) ),
+		"Content-Type: $mime_type",
+	);
 
 	echo $pre_output . $output;
 
-	return ob_get_clean();
+	return array(
+		'headers' => $headers,
+		'content' => ob_get_clean(),
```

Comment:
Nice. "content" is better than "output" which is too generic and could describe the headers as well.